From d1e0abf8f0184d29d37ea1769aa8c372b5fae619 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ren=C3=A9=20Sass?= <mail@renesass.de>
Date: Thu, 3 Feb 2022 11:23:56 +0100
Subject: [PATCH] Updated dependency comparison

---
 smac/utils/dependencies.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/smac/utils/dependencies.py b/smac/utils/dependencies.py
index 5a8bc3698..5c1c0a654 100644
--- a/smac/utils/dependencies.py
+++ b/smac/utils/dependencies.py
@@ -2,7 +2,7 @@
 import pkg_resources
 import re
 import typing
-from distutils.version import LooseVersion
+from packaging.version import Version
 
 __copyright__ = "Copyright 2021, AutoML.org Freiburg-Hannover"
 __license__ = "3-clause BSD"
@@ -42,18 +42,18 @@ def verify_packages(packages: typing.Union[typing.List[str], str]) -> None:
 def _verify_package(name: str, operation: str, version: str) -> None:
     try:
         distribution = pkg_resources.get_distribution(name)
-        installed_version = LooseVersion(distribution.version)
+        installed_version = Version(distribution.version)
     except pkg_resources.DistributionNotFound:
         try:
             module = importlib.import_module(name)
-            installed_version = LooseVersion(module.__version__)  # type: ignore[attr-defined] # noqa F821
+            installed_version = Version(module.__version__)  # type: ignore[attr-defined] # noqa F821
         except ImportError:
             raise MissingPackageError(name)
 
     if not operation:
         return
 
-    required_version = LooseVersion(version)
+    required_version = Version(version)
 
     if operation == "==":
         check = required_version == installed_version
@@ -97,9 +97,9 @@ class IncorrectPackageVersionError(Exception):
     def __init__(
         self,
         package_name: str,
-        installed_version: LooseVersion,
+        installed_version: Version,
         operation: str,
-        required_version: LooseVersion,
+        required_version: Version,
     ) -> None:
         self.package_name = package_name
         self.installed_version = installed_version
